APMrover2/AP_Arming.cpp

```diff
@@ -81,6 +81,10 @@ bool AP_Arming_Rover::gps_checks(bool display_failure)
 
 bool AP_Arming_Rover::pre_arm_checks(bool report)
 {
+    //are arming checks disabled?
+    if (checks_to_perform == ARMING_CHECK_NONE) {
+        return true;
+    }
```

Comment:
I think E-Stop should take priority over no arming checks. Otherwise you can arm in to a E-stopped state and nothing will work. I think this is also issue for plane.

Reply:
I'm not going to disagree with you - and whatever we decide should be the same across all vehicles.

I am going to say I'm trying to minimise actual behavioural changes with this PR - so I'd prefer not to make any change here.